backend/src/main/kotlin/blue/mild/covid/vaxx/service/PatientService.kt

```diff
@@ -1,36 +1,101 @@
 import java.util.UUID
 
 class PatientService {
-    suspend fun getAllPatients(): List<PatientDto> = newSuspendedTransaction {
-        Patient.selectAll().map {
-            PatientDto(
-                id = it[Patient.id].toUuid(),
-                firstName = it[Patient.firstName],
-                lastName = it[Patient.lastName],
-                personalNumber = it[Patient.personalNumber],
-                phoneNumber = it[Patient.phoneNumber],
-                email = it[Patient.email]
-            )
-        }
+    suspend fun getPatientById(patientId: UUID): PatientDtoOut = newSuspendedTransaction {
+        val query = Patient
+            .leftJoin(Answer, { id }, { Answer.patientId })
+            .select { Patient.id eq patientId.toString() }
+        val answers = query
+            .map { AnswerDto(it[Answer.questionId].toUuid(), it[Answer.value]) }
+
+        query
+            .firstOrNull()
+            ?.let {
+                PatientDtoOut(
+                    id = it[Patient.id].toUuid(),
+                    firstName = it[Patient.firstName],
+                    lastName = it[Patient.lastName],
+                    personalNumber = it[Patient.personalNumber],
+                    phoneNumber = it[Patient.phoneNumber],
+                    email = it[Patient.email],
+                    answers = answers
+                )
+            } ?: throw entityNotFound<Patient>(Patient::id, patientId)
     }
 
-    suspend fun savePatient(patientDto: NewPatientDto) = newSuspendedTransaction {
+    suspend fun getAllPatients(): List<PatientDtoOut> =
+        newSuspendedTransaction { getAndMapPatients() }
+
+    suspend fun getPatientsByPersonalNumber(patientPersonalNumber: String): List<PatientDtoOut> =
+        newSuspendedTransaction { getAndMapPatients { Patient.personalNumber eq patientPersonalNumber } }
+
+    suspend fun getPatientsByEmail(email: String): List<PatientDtoOut> =
+        newSuspendedTransaction { getAndMapPatients { Patient.email eq email } }
+
+    suspend fun savePatient(patientDto: PatientRegistrationDtoIn) = newSuspendedTransaction {
+        val patientId = UUID.randomUUID().toString()
+
         Patient.insert {
-            it[id] = UUID.randomUUID().toString()
+            it[id] = patientId
             it[firstName] = patientDto.firstName
             it[lastName] = patientDto.lastName
             it[personalNumber] = patientDto.personalNumber
             it[phoneNumber] = patientDto.phoneNumber
             it[email] = patientDto.email
         }
+
+        Answer.batchInsert(patientDto.answers) {
+            this[Answer.patientId] = patientId
+            this[Answer.questionId] = it.questionId.toString()
+            this[Answer.value] = it.value
+        }
+    }
+
+    suspend fun deletePatientById(patientId: UUID) = newSuspendedTransaction {
+        val count = Patient.deleteWhere { Patient.id eq patientId.toString() }
+        if (count == 1) PatientDeletedDtoOut(true)
```

Comment:
Nechceme vsude psat `if {} else {}`?

Reply:
no v tomhle pripade je to nahrada ternarniho operatoru, takze tady spis ne, obecne ano